### PR TITLE
Added type hints for Pytorch Marian calls

### PR DESCRIPTION
**Pull request by clefourrier**
As the title suggests, this PR adds type hints to the PyTorch Marian calls. (#16059 )

@Rocketknight1

### CODE REVIEW

File: src/transformers/models/marian/modeling_marian.py
Comment (patil-suraj) [inline src/transformers/models/marian/modeling_marian.py:79]: same comment as above.

File: src/transformers/models/marian/modeling_marian.py
Comment (clefourrier) [reply src/transformers/models/marian/modeling_marian.py:62]: I will remove it from this PR then

File: src/transformers/models/marian/modeling_marian.py
Comment (clefourrier) [reply src/transformers/models/marian/modeling_marian.py:79]: same

File: src/transformers/models/marian/modeling_marian.py
Comment (clefourrier) [reply src/transformers/models/marian/modeling_marian.py:95]: same

**Comment by clefourrier** [comment]
Hm, do I need to change the BART too for consistency?  
```
src/transformers/models/marian/modeling_marian.py: copy does not match models.bart.modeling_bart.BartAttention at line 144
src/transformers/models/marian/modeling_marian.py: copy does not match models.bart.modeling_bart.BartEncoderLayer at line 289
src/transformers/models/marian/modeling_marian.py: copy does not match models.bart.modeling_bart.BartDecoderLayer at line 356
src/transformers/models/marian/modeling_marian.py: copy does not match models.bart.modeling_bart.BartForCausalLM at line 1529
src/transformers/models/pegasus/modeling_pegasus.py: copy does not match models.marian.modeling_marian.MarianSinusoidalPositionalEmbedding at line 109
src/transformers/models/roformer/modeling_roformer.py: copy does not match models.marian.modeling_marian.MarianSinusoidalPositionalEmbedding at line 73
Run `make fix-copies` or `python utils/check_copies.py --fix_and_overwrite` to fix them.
```

Running `make fix-copies` basically reverts all my changes.

**Comment by Rocketknight1** [comment]
Hi @clefourrier, sorry for the delay! We will indeed need to make changes to BART to make the tests work - but we already have an existing BART PR here: https://github.com/huggingface/transformers/pull/16270. Probably the easiest thing to do is:

1. Wait until we merge that PR
2. If any changes in Marian are still needed, rebase your current PR onto it (make sure to force push afterwards!)
3. Check tests again afterwards

**Comment by clefourrier** [comment]
Hi @Rocketknight1 , perfect! Can you ping me once the BART PR is merged then?

**Comment by Rocketknight1** [comment]
@clefourrier It was just merged about half an hour ago! You'll probably need to pull changes from upstream to the master branch of your fork, and then rebase (with force push!) onto that.

**Comment by clefourrier** [comment]
@Rocketknight1 Well, it seems the last failures comes from children models (pegasus and roformer) - do they have their own PR?

File: src/transformers/models/marian/modeling_marian.py
Comment (Rocketknight1) [inline src/transformers/models/marian/modeling_marian.py:480]: A lesser contributor would have just given up and written `Any` for this one. I respect it.

**Review by Rocketknight1 — APPROVED** [review]
PR looks great! The issue with Pegasus and Roformer was just that they copied the sinusoidal embeddings from `Marian`. Running `make fix-copies` after the rebase fixed the issue, so this should be ready to merge once the tests are green!

**Pull request merged**
